utils/check_availability.js

```diff
@@ -100,7 +100,7 @@ async function checkRangeAvailability(fromRevision, toRevision, stopWhenAllAvail
  * @return {boolean}
  */
 async function checkAndDrawRevisionAvailability(table, name, revision) {
-  const promises = fetcherOptions.map(platform => browserFetcher.canDownload(name, revision, platform));
+  const promises = fetcherOptions.map(platform => browserFetcher.canDownload({ name, revision, download: true }, platform));
```

Comment:
the `download: true` argument to the `canDownload` method does look uncanny to me.

How do you feel splitting `download` bit from BrowserDescriptor into a separate field? It'll solve all the weird typing issues since absolute majority of clients of `BrowserDescriptor` object (e.g. `revisionURL()` and `canDownload()` functions) don't need `download` bit.

For example, `browsers.json` could look like this:

```json
{
  "comment": "Do not edit this file, use utils/roll_browser.js",
  "download": ["chromium", "firefox", "webkit"],
  "browsers": [
    {
      "name": "chromium",
      "revision": "791201"
    },
    {
      "name": "firefox",
      "revision": "1140"
    },
    {
      "name": "webkit",
      "revision": "1317"
    }
  ]
}
```

If you like the idea but don't have time to follow-up with it, I'll send you a PR as this lands.